src/main/java/TaskList.java

```diff
@@ -0,0 +1,90 @@
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+public class TaskList {
+    Task[] taskList = new Task[100];
+    // The first index that is empty
+    private int firstEmptyIndex = 0;
+
+    public void addTask(String input, Task.TaskType taskType, boolean isDone) throws IOException {
+        Task task;
+        try {
+            task = taskType == Task.TaskType.ToDo
+                    ? new Todo(input, isDone)
+                    : taskType == Task.TaskType.Event
+                    ? new Event(input, isDone)
+                    : new Deadline(input, isDone);
+        } catch (DateTimeParseException e) {
+            return;
+        }
+
+        taskList[firstEmptyIndex] = task;
+        DataReadWriter.saveAddedTask(task);
+
+        String msg = "Got it. I've added this task:\n" + "  "
+                + taskList[firstEmptyIndex].toString()
+                + "\n" + "Now you have " + (firstEmptyIndex + 1) + " tasks in the list";
+        System.out.println(msg);
+        ++firstEmptyIndex;
+    }
+
+    public void appendLoadedTask(Task task) throws IOException {
+
+        taskList[firstEmptyIndex] = task;
+        ++firstEmptyIndex;
+    }
+
+    public void markTaskAsDone(int taskNum) {
+        if (taskNum >= firstEmptyIndex) {
+            System.out.println("There is no task with index " + (taskNum + 1));
+            return;
+        }
+        taskList[taskNum].markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(taskList[taskNum].toString());
+    }
+
+    public void markTaskAsUnDone(int taskNum) {
+        if (taskNum >= firstEmptyIndex) {
+            System.out.println("There is no task with index " + (taskNum + 1));
```

Comment:
There are some similar cases where you print something directly from this method. Would it be better if you use Ui class to ensure all the statements are at the same level?